sdk/core/AzureCore/Source/DataStructures/Collections.swift

```diff
@@ -246,16 +246,12 @@ public class PagedCollection<SingleElement: Codable> {
                 }
             }
             if let returnError = returnError {
-                DispatchQueue.main.async {
-                    completionHandler(.failure(returnError))
-                }
+                completionHandler(.failure(returnError))
```

Comment:
Should the completionHandler be called inside the `queue` (either the DispatchQueue.main or the queue in the ClientOption)?

Reply:
This caused the demo test to hang, even when called on the right dispatch queue. Removing this entirely caused the demo test to work correctly and still works correctly with the AzureSwiftSDKDemo.

Reply:
Thanks. Do you know why removing this will cause the demo app to work, but keeping it inside the queue failed? Maybe add comments to explain why calling completionHandler out of DispatchQueue will work, but keeping them inside does not work.